### PR TITLE
Add `cycle-lists-with-keyboard-shortcuts` feature

### LINKED ISSUE #1450: Support pressing arrow-up when choosing labels
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/refined-github/issues/1450)

It's new feature proposal.

Scenario:
I created pull request, and I'm adding labels. I want to add label `Please CR: JS`. I type first two letters (`Pl`) into search input, and now I see 5 similar labels (e.g. `Please CR: CSS`, `Please CR: HTML`).  The one that I'm looking for is at the very end. I'd like to navigate to this label by pressing `up` arrow. Unfortunately, nothing happens.

Feature proposal:
During label searching, pressing `up` arrow at first item navigates me to the last one, and pressing `down` arrow at last item navigates me to the first item.

> There is a $40.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/refined-github/issues/1450).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/refined-github/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

### PR DESCRIPTION
**Pull request by notlmn**
Closes #1450.

- Makes use of custom events emitted by GitHub to detect and add events to input elements.
- As opposed to #2008, this PR adds `keydown` event listeners to only search fields, which is also what GitHub uses (you cannot use up and down arrow keys to navigate when the search field is not focused).
- Supports cycling through labels, projects, milestones and all other sidebar elements, even when a search filter is applied.

## Preview

labels-cycling

## Test
Can be tested on any issue or PR that you have permissions to edit.

### CODE REVIEW

File: source/features/cycle-lists-with-keyboard-shortcuts.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		populateSelectableItems();
+
+		const input = event.delegateTarget as HTMLElement;
+		input.addEventListener('keydown', handleKeyDown);
+	});
+}
+
+features.add({
+	id: 'cycle-lists-with-keyboard-shortcuts',
+	description: 'Cycle "select lists" (labels, etc) when selecting them with the `↑` and `↓` keys.',
```
Comment (sindresorhus) [inline source/features/cycle-lists-with-keyboard-shortcuts.tsx]:
```suggestion
	description: 'Cycle "popover lists" (labels, milestones, etc) when selecting them with the `↑` and `↓` keys.',
```

?

**Comment by sindresorhus** [comment]
The functionality seems to work well. Needs to be mentioned in the readme.

**Comment by notlmn** [comment]
> Needs to be mentioned in the readme.

Done!

File: source/features/cycle-lists-with-keyboard-shortcuts.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		populateSelectableItems();
+
+		const input = event.delegateTarget as HTMLElement;
+		input.addEventListener('keydown', handleKeyDown);
+	});
+}
+
+features.add({
+	id: 'cycle-lists-with-keyboard-shortcuts',
+	description: 'Cycle "popover lists" (labels, milestones, etc) when selecting them with the `↑` and `↓` keys.',
```
Comment (sindresorhus) [inline source/features/cycle-lists-with-keyboard-shortcuts.tsx]: Can you include the `screenshot` property here.

Comment (notlmn) [reply source/features/cycle-lists-with-keyboard-shortcuts.tsx]: Done!

**Pull request merged**